cpp/src/arrow/array.h

```diff
@@ -39,7 +40,7 @@ class Array {
   Array(const std::shared_ptr<DataType>& type, int32_t length, int32_t null_count = 0,
       const std::shared_ptr<Buffer>& null_bitmap = nullptr);
 
-  virtual ~Array() {}
+  virtual ~Array() = default;
```

Comment:
Aside: I tried to find definitive documentation on when to use `= default` vs `= 0` in C++11, if you have any pointers